src/CaptureFile/CaptureFileOutputStream.cpp

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "CaptureFile/CaptureFileOutputStream.h"
+
+#include <absl/base/casts.h>
+#include <google/protobuf/io/coded_stream.h>
+#include <google/protobuf/io/zero_copy_stream_impl.h>
+
+#include <optional>
+#include <string>
+
+#include "CaptureFileConstants.h"
+#include "OrbitBase/File.h"
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/SafeStrerror.h"
+
+namespace orbit_capture_file {
+
+namespace {
+
+class CaptureFileOutputStreamImpl final : public CaptureFileOutputStream {
+ public:
+  explicit CaptureFileOutputStreamImpl(std::filesystem::path path) : path_{std::move(path)} {}
+  ~CaptureFileOutputStreamImpl() noexcept override;
+
+  ErrorMessageOr<void> Initialize();
```

Comment:
I guess this is also explicitly `[[nodiscard]]`? Nonetheless, I'd add it as the method below has it.

Reply:
done. (interestingly I accidentally dropped implicit nodiscard - fixed in #2035)